### ISSUE TITLE
Fix S4275 false positive when field in property is set through a method.

### ISSUE THREAD

**Issue by mary-georgiou-sonarsource** [SonarSource/sonar-dotnet#5259]
### Description
FP is raised by S4275 when the field in a property is set through a method.
sonar-dotnet version: 8.33

### Repro steps

```csharp
        private int[] attributes;
        public int[] Attributes
        {
            set { value.CopyTo(attributes, 0); } // Noncompliant - FP
        }
```

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1869](https://sonarsource.atlassian.net/browse/NET-1869)

[NET-1869]: https://sonarsource.atlassian.net/browse/NET-1869?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
